src/data/finsupp/basic.lean

```diff
@@ -1212,36 +1212,44 @@ lemma prod_inv [has_zero M] [comm_group G] {f : α →₀ M}
 finset.sum_sub_distrib
 
 @[to_additive]
-lemma prod_add_index [add_comm_monoid M] [comm_monoid N] {f g : α →₀ M}
+lemma prod_add_index [add_zero_class M] [comm_monoid N] {f g : α →₀ M}
+  {h : α → M → N} (h_zero : ∀ a ∈ f.support ∪ g.support, h a 0 = 1)
+  (h_add : ∀ a b₁ b₂, h a (b₁ + b₂) = h a b₁ * h a b₂) :
+  (f + g).prod h = f.prod h * g.prod h :=
+begin
+  rw [finsupp.prod_of_support_subset f (subset_union_left _ g.support) h h_zero,
+      finsupp.prod_of_support_subset g (subset_union_right f.support _) h h_zero,
+      ←finset.prod_mul_distrib,
+      finsupp.prod_of_support_subset (f + g) finsupp.support_add h h_zero],
+  exact finset.prod_congr rfl (λ x hx, (by apply h_add)),
```

Comment:
Does it make sense to also weaken to `h_add : ∀ (a ∈ f.support ∪ g.support) b₁ b₂, ...`?

Reply:
Ah yes, that works, thanks!  I've made that change and now I'm waiting to see what breaks.  :)